examples/showcase/instancing/app.ts

```diff
@@ -211,9 +211,9 @@ export default class AppAnimationLoopTemplate extends AnimationLoopTemplate {
   
     this.cube = new InstancedCube(device, {
       bindings: {
-        appUniforms: this.uniformStore.getManagedUniformBuffer(device, 'app'),
-        dirlightUniforms: this.uniformStore.getManagedUniformBuffer(device, 'dirlightMaterial'),
-        pickingUniforms: this.uniformStore.getManagedUniformBuffer(device, 'colorPicking'),
+        app: this.uniformStore.getManagedUniformBuffer(device, 'app'),
+        dirlight: this.uniformStore.getManagedUniformBuffer(device, 'dirlight'),
```

Comment:
If the uniform block name always matches the module name, it should be possible to just "bind uniforms from this store" instead of explicitly specify each name?

The current pattern does not work with nested module dependencies, e.g. shader module A depends on B, B depends on C - when the user adds module A to the model, they have to manually add all of  `a`, `b` and `c` to bindings.

Reply:
> The current pattern does not work with nested module dependencies, e.g. shader module A depends on B, B depends on C - when the user adds module A to the model, they have to manually add all of a, b and c to bindings. 

In some sense, that is not much different from today. Unless someone is just using the defaults from a nested module, they do need to be aware of that nested module and start setting uniform values.

Additionally, if we add those nested modules automatically I will have to reduce the strict type safety in the uniformStore class. I can't enforce a valid module key is used, and I can't check typing for non-declared modules. 

Before we worry too much about nesting, which are our main nested modules? We have the lights settings in the lighting modules, which we probably would rather declare explicitly so we can enjoy the typings. There are some small util modules like `random` but they are not a problem as they don't expose uniforms.

Reply:
> If the uniform block name always matches the module name, it should be possible to just "bind uniforms from this store" instead of explicitly specify each name?

Perhaps. I am still building intuition around this so I hesitate to move too fast.
- Initially I was imagining that a single uniform store could handle multiple models, and that models would need multiple uniformStores. Some uniform buffers are shared between models and should only be created once etc. I haven't yet been able to build a complex example like that. The restored glTF example might be a good test, as it will share projection buffer over all models, material buffers between some models etc.
- Another concern is that the uniformStore so far has only been used with WebGL examples, and convention may not work in WebGPU. All the WebGPU examples broke when we refactored the ShaderLayout, I am working on repairing those.

Reply:
> In some sense, that is not much different from today. Unless someone is just using the defaults from a nested module, they do need to be aware of that nested module and start setting uniform values.

In v8, the user never manually sets module uniforms. They update module settings, which are documented, then `module.getUniforms()` is called internally along the dependency chain and returns a merged dictionary of key value pairs.

> which are our main nested modules?

deck.gl `project32` / `project`